src/Lachain.Consensus/HoneyBadger/HoneyBadgerId.cs

```diff
@@ -13,16 +19,16 @@ protected bool Equals(HoneyBadgerId other)
             return Era == other.Era;
         }
 
-        public bool Equals(IProtocolIdentifier other)
+        public bool Equals(IProtocolIdentifier? other)
```

Comment:
Looks like some contradiction together with next line: in this line we tell that other can be null,  but in next line we tell that it can't be null. Why we fix it here at all?